packages/common/client-utils/src/storageUtils.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IPartialSnapshotWithContents } from "@fluidframework/driver-definitions";
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+
+/**
+ * Utility API to check if the type of snapshot contents is `IPartialSnapshotWithContents`.
+ * @alpha
+ * @param obj - obj whose type needs to be identified.
+ */
+export function instanceOfIPartialSnapshotWithContents(
+	// eslint-disable-next-line @rushstack/no-new-null
+	obj: ISnapshotTree | IPartialSnapshotWithContents | null | undefined,
+): obj is IPartialSnapshotWithContents {
+	return (
+		obj !== null &&
+		obj !== undefined &&
+		"couldBePartialSnapshot" in obj &&
```

Comment:
I'd replace first two checks to typeof obj === "object" && obj !== null.
It's stronger, at it gives you only objects (no strings, no numbers, etc.). TS type system should ideally not allow such types here, but it costs you nothing to be more strict.

Reply:
And once you know it's object, you can skill 3rd line and directly test property - if it's not there, it will evaluate to undefined

Reply:
Changed the check but you cannot check prop that directly as it is not present in ISnapshotTree